src/workerd/api/streams/common.h

```diff
@@ -200,7 +200,14 @@ class WritableStreamSink {
   virtual kj::Maybe<kj::Promise<DeferredProxy<void>>> tryPumpFrom(
       ReadableStreamSource& input, bool end);
 
-  virtual void abort(kj::Exception reason) = 0;
+  enum class AbortOption {
+    NONE,
+    // Instructs the sink to reject pending writes when aborting. This is used when
+    // the InternalWritableStreamAbortClearsQueue flag is set.
+    DRAIN,
```

Comment:
Hmm, I usually associate the word "drain" with "gracefully complete all pending requests before shutting down", but here it seems to have the opposite meaning?

Maybe `FINISH_QUEUED_WRITES` and `CANCEL_QUEUED_WRITES` would be clearer names?